test/240_dns_add_name_test.sh

```diff
@@ -0,0 +1,24 @@
+#! /bin/bash
+
+. ./config.sh
+
+C1=10.2.0.78
+C2=10.2.0.34
+NAME=seetwo.weave.local
+
+start_suite "Add and remove names on a single host"
+
+weave_on $HOST1 launch-dns 10.2.254.1/24
+
+weave_on $HOST1 run $C2/24 -t --name=c2 ubuntu
+weave_on $HOST1 run --with-dns $C1/24 -t --name=c1 aanand/docker-dnsutils /bin/sh
+
+weave_on $HOST1 dns-add $C2/24 c2 -h $NAME
+
+assert_dns_record $HOST1 c1 $NAME $C2
+
+weave_on $HOST1 dns-remove $C2/24 c2
+
+assert_raises "exec_on $HOST1 c1 getent hosts $C2 | grep $NAME"
```

Comment:
`assert_raises` checks whether the exist status matches $2, defaulting to 0. So the above actually checks whether the reverse lookup of the IP succeeds. Which, surprisingly, it does! Why? Because it got cached, since `assert_dns_record` performs both forward and reverse lookups, and weavedns caches answers.

So we need a better test. I suggest adding a 2nd record, removing it again, and then check that a lookup fails. NB: the check should be done with a *forward* lookup; reverse lookups are expensive when they fail since they go to upstream dns server.